index.js

```diff
@@ -10,86 +10,129 @@ if ("SERVER_LOGGER" in process.env) {
   }
 }
 
-const fastify = require("fastify")({ logger: logger })
+const axios = require("axios");
 
-// EXIT IF POSTGRES_CONNECTION ENV VARIABLE NOT SET
-if (!("POSTGRES_CONNECTION" in process.env)) {
-  throw new Error("Required ENV variable POSTGRES_CONNECTION is not set. Please see README.md for more information.");
-}
+async function build() {
+  const fastify = require("fastify")({ logger: logger });
+  await fastify.register(require('@fastify/express'));
+  const queryString = require("query-string");
+  fastify.use((req, res, done) => {
+    reqParams = req.url.split("?")[1];
+    const parsedReqParams = queryString.parse(reqParams);
+    const formId = parsedReqParams.form_id;
+    tempToken = parsedReqParams.temp_token;
+    if (formId) {
+      axios
+        .get(`${process.env.FORMS_ENDPOINT}${formId}.json`, {
+          headers: tempToken && tempToken.length > 0 ? {
+            Authorization: `TempToken ${tempToken}`,
+          } : {},
+        })
+        .then((res) => {
+          if (res && res.status === 200) {
+            done();
+          } else {
+            done("Forbidden");
+          }
+        })
+        .catch((error) => {
+          console.log(error)
+          done(error.detail);
+        });
+    } else if ("/health-check" == req.url) {
+      done()
+    } else {
+      done("Authentication Failure");

```

Comment:
Can we also add a status code (401)

Reply:
I couldn't find how to do this from fastify documentation